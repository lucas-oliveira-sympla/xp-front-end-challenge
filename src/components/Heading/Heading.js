import React from 'react';
import { node } from 'prop-types';

import HeadingStyled from './Heading.styles';

const propTypes = {
  children: node.isRequired,
};

const Heading = ({ children }) => <HeadingStyled>{children}</HeadingStyled>;

Heading.propTypes = propTypes;

export default Heading;
