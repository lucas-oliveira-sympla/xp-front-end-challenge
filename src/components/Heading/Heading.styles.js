import styled from 'styled-components';
import breakpoint from 'styled-components-breakpoint';

import colors from '../../theme/colors';

const HeadingStyled = styled.h1`
  color: ${colors.primary[700]};
  display: inline-block;
  width: 100%;
  padding: 0;
  font-family: 'Roboto', sans-serif;
  font-weight: normal;
  font-size: 18px;

  ${breakpoint('desktop')`
    font-size: 24px;
  `}
`;

export default HeadingStyled;
