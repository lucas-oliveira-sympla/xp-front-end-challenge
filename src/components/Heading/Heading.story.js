import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import colors from '../../theme/colors';

import Heading from './Heading';

const BackgroundStyled = styled.div`
  background-color: ${colors.primary[900]};
  padding: 5%;
`;

const stories = storiesOf('Heading', module);
stories.add('default', () => (
  <BackgroundStyled>
    <Heading>√Ålbuns recentemente buscados</Heading>
  </BackgroundStyled>
));
