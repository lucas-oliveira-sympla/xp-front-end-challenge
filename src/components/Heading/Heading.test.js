import React from 'react';
import { mount } from 'enzyme';

import Heading from './Heading';

describe('<Heading />', () => {
  let wrapper;

  describe('with default props', () => {
    beforeAll(() => {
      wrapper = mount(<Heading>Test heading</Heading>);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
