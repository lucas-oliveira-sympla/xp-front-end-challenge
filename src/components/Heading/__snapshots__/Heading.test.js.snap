// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> with default props should match snapshot 1`] = `
<Heading>
  <Headingstyles__HeadingStyled>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Headingstyles__HeadingStyled-sc-15dfmro-0",
            "isStatic": false,
            "lastClassName": "depeJO",
            "rules": Array [
              "color:",
              "#FAFAFA",
              ";display:inline-block;width:100%;padding:0;font-family:'Roboto',sans-serif;font-weight:normal;font-size:18px;",
              [Function],
            ],
          },
          "displayName": "Headingstyles__HeadingStyled",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Headingstyles__HeadingStyled-sc-15dfmro-0",
          "target": "h1",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h1
        className="Headingstyles__HeadingStyled-sc-15dfmro-0 depeJO"
      >
        Test heading
      </h1>
    </StyledComponent>
  </Headingstyles__HeadingStyled>
</Heading>
`;
